### PR TITLE
feat(DMS): add resource kafka consumer group

### PR DESCRIPTION
**Pull request by zhangting951007**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```  
add resource kafka consumer group
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dms/ TESTARGS='-run TestAccDmsKafkaConsumerGroup_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsKafkaConsumerGroup_basic -timeout 360m -parallel 4
=== RUN   TestAccDmsKafkaConsumerGroup_basic
=== PAUSE TestAccDmsKafkaConsumerGroup_basic
=== CONT  TestAccDmsKafkaConsumerGroup_basic
--- PASS: TestAccDmsKafkaConsumerGroup_basic (799.36s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       799.398s
```

### CODE REVIEW

File: docs/resources/dms_kafka_consumer_group.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  Changing this parameter will create a new resource.
+
+* `name` - (Required, String, ForceNew) Specifies the name of the consumer group.
+
+  Changing this parameter will create a new resource.
+
+* `description` - (Optional, String) Specifies the description of the consumer group.
+
+* `state` - (Optional, String) Indicates the state of the consumer group.
+  Value options: **DEAD**, **EMPTY**, **PreparingRebalance**, **CompletingRebalance**, **Stable**.
```
Comment (houpeng80) [inline docs/resources/dms_kafka_consumer_group.md]:
```suggestion
* `state` - Indicates the state of the consumer group. The alue can be: **DEAD**, **EMPTY**, **PreparingRebalance**, **CompletingRebalance**, **Stable**.
```
and fix the other attributes of coordinator_id, lag, created_at

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		getKafkaConsumerGroupHttpUrl = "v2/{project_id}/instances/{instance_id}/groups/{group}"
+		getKafkaConsumerGroupProduct = "dms"
+	)
+	getKafkaConsumerGroupClient, err := cfg.NewServiceClient(getKafkaConsumerGroupProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DMS Client: %s", err)
+	}
+
+	// Split instance_id and group from resource id
+	parts := strings.SplitN(state.Primary.ID, "/", 2)
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]:
```suggestion
	parts := strings.Split(state.Primary.ID, "/")
```
and fix other places

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testDmsKafkaConsumerGroup_basic(name string) string {
+	return fmt.Sprintf(`
+%s
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go:119]: add a new line

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  instance_id = huaweicloud_dms_kafka_instance.test.id
+  name        = "%s"
+  description = "add"
+}
+`, testAccKafkaInstance_compatible(name), name)
+}
+
+func testDmsKafkaConsumerGroup_basic_update(name string) string {
+	return fmt.Sprintf(`
+%s
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go:131]: ditto

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+			"lag": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Specifies the lag number of the consumer group.`,
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `Specifies the created time of the consumer group.`,
+			},
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]:
```suggestion
			"state": {
				Type:        schema.TypeString,
				Computed:    true,
				Description: `Indicates the state of the consumer group.`,
			},
			"coordinator_id": {
				Type:        schema.TypeInt,
				Computed:    true,
				Description: `Indicates the coordinator id of the consumer group.`,
			},
			"lag": {
				Type:        schema.TypeInt,
				Computed:    true,
				Description: `Indicates the lag number of the consumer group.`,
			},
			"created_at": {
				Type:        schema.TypeInt,
				Computed:    true,
				Description: `Indicates the created time of the consumer group.`,
			},
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+			return diag.Errorf("error updating DMS Kafka consumer group: %s", err)
+		}
+	}
+
+	return resourceDmsKafkaConsumerGroupRead(ctx, d, meta)
+}
+
+func buildUpdateKafkaConsumerGroupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"group_name": utils.ValueIngoreEmpty(d.Get("name")),
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]:
```suggestion
		"group_name": d.Get("name"),
```

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+
+	deleteKafkaConsumerGroupOpt.JSONBody = utils.RemoveNil(buildBatchDeleteKafkaConsumerGroupBodyParams(d, cfg))
+	_, err = deleteKafkaConsumerGroupClient.Request("POST", deleteKafkaConsumerGroupPath, &deleteKafkaConsumerGroupOpt)
+	if err != nil {
+		return diag.Errorf("error deleting DMS Kafka consumer group: %s", err)
+	}
+	return resourceDmsKafkaConsumerGroupRead(ctx, d, meta)
+}
+
+func buildBatchDeleteKafkaConsumerGroupBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (houpeng80) [inline huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: delete the not used param

File: docs/resources/dms_kafka_consumer_group.md (hunk 1 above)
Comment (zhangting951007) [reply docs/resources/dms_kafka_consumer_group.md]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 2 above)
Comment (zhangting951007) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 5 above)
Comment (zhangting951007) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 6 above)
Comment (zhangting951007) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go (hunk 7 above)
Comment (zhangting951007) [reply huaweicloud/services/dms/resource_huaweicloud_dms_kafka_consumer_group.go]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 4 above)
Comment (zhangting951007) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go:131]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go (hunk 3 above)
Comment (zhangting951007) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_kafka_consumer_group_test.go:119]: done

**Pull request merged**